packages/terra-action-header/src/ActionHeader.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import ResponsiveElement from 'terra-responsive-element';
+import 'terra-base/lib/baseStyles';
+import ActionHeaderContainer from './ActionHeaderContainer';
+import styles from './ActionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Text to be displayed as the title in the header bar.
+   */
+  title: PropTypes.string,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=1`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  /**
+   * Callback function for when the close button is clicked. The back button will not display if this is not set.
+   * On small viewports a back button will be displayed instead of a close button when a separate onBack callback is not set.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function for when the back button is clicked. The back button will not display if this is not set.
+   */
+  onBack: PropTypes.func,
+  /**
+   * A Boolean indicating if close button should be displayed on small viewports when separate onBack callback is not set.
+   */
+  keepCloseButton: PropTypes.bool,
+  /**
+   * Child element to be displayed on the right end of the header.
+   */
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  title: '',
+  level: 1,
+  onClose: null,
+  onBack: null,
+  keepCloseButton: false,
+  children: null,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+const ActionHeader = ({
+  title,
+  level,
+  onClose,
+  onBack,
+  keepCloseButton,
+  children,
+  ...customProps }, {
+  intl,
+}) => {
+  const attributes = Object.assign({}, customProps);
+  const backText = intl.formatMessage({ id: 'Terra.actionHeader.back' });
+  const closeText = intl.formatMessage({ id: 'Terra.actionHeader.close' });
+
+  const closeButton = onClose ? <Button variant="utility" isIconOnly icon={<IconClose />} text={closeText} onClick={onClose} /> : null;
+  const backButton = onBack ? <Button variant="utility" isIconOnly icon={<IconLeft />} text={backText} onClick={onBack} /> : null;
+
+  let closeButtonSmall;
+  let backButtonSmall;
+  if (onClose && !onBack && !keepCloseButton) {
+    backButtonSmall = <Button variant="utility" isIconOnly icon={<IconLeft />} text={backText} onClick={onClose} />;
+    closeButtonSmall = null;
+  } else {
+    closeButtonSmall = closeButton;
+    backButtonSmall = backButton;
+  }
+
+  const leftButtonsDefault = (
+    <div className={cx('left-buttons')}>
+      {backButton}
+    </div>
+  );
+
+  const rightButtonsDefault = (
```

Comment:
Lets check if `closeButton` is null, otherwise we are adding an empty 'right-buttons' div

Reply:
Updated to check for null on both close and back buttons [here](https://github.com/cerner/terra-core/commit/b1e09f51f44a1123922961a673cc07c9e2925f33).